wallet/src/de/schildbach/wallet/ui/WalletTransactionsFragment.java

```diff
@@ -378,6 +376,23 @@ protected CharSequence collectWalletDump() {
         popupMenu.show();
     }
 
+    @Override
+    public void onTransactionRowClicked(Transaction tx) {
+        Direction direction = tx.getValue(wallet).signum() < 0 ? Direction.SENT : Direction.RECEIVED;
+        String address;
+        if (direction == Direction.SENT) {
+            address = WalletUtils.getToAddressOfSent(tx, wallet).toBase58();
```

Comment:
`WalletUtils.getToAddressOfSent` returns `null` for internal transactions.  This method looks for the first output in the transaction that doesn't belong to this wallet.  For an internal transaction, all inputs are from the wallet and all outputs are from the wallet.  So this returns `null`.

We will need some handling of internal transactions, not only for this but for the display on the screen.  I am encountering the same issue on the Transaction Status Story.

I would be OK to merge this story as is, and allow my Transaction Status story to handle this case of internal transactions.

```
2019-11-15 18:00:18.858 31460-31460/hashengineering.dash.wallet.beta E/AndroidRuntime: FATAL EXCEPTION: main
    Process: hashengineering.dash.wallet.beta, PID: 31460
    java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String org.bitcoinj.core.Address.toBase58()' on a null object reference
        at de.schildbach.wallet.ui.WalletTransactionsFragment.onTransactionRowClicked(WalletTransactionsFragment.java:384)
        at de.schildbach.wallet.ui.TransactionsAdapter$1.onClick(TransactionsAdapter.java:248)
        at android.view.View.performClick(View.java:6294)
        at android.view.View$PerformClick.run(View.java:24770)
        at android.os.Handler.handleCallback(Handler.java:790)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

Reply:
@HashEngineering I appreciate if you fix this in your Transaction Status PR.

Reply:
It will be fixed in a different PR to modify this Transaction Details Popup to include other status messages.